Taken from upstream commit 5bd23bd4a481932ce458eacba865ece1c6c72211
"Update build to support Java 25"

REMOVE THIS PATCH ON NEXT UPDATE

--- buildSrc/build.gradle.orig	2026-02-16 06:34:26
+++ buildSrc/build.gradle	2026-09-09 15:51:49
@@ -8,6 +8,6 @@
 
 dependencies {
     implementation 'com.diffplug.spotless:spotless-plugin-gradle:8.1.0'
-    implementation 'net.ltgt.gradle:gradle-errorprone-plugin:4.0.0'
+    implementation 'net.ltgt.gradle:gradle-errorprone-plugin:4.3.0'
     implementation 'de.obqo.decycle:de.obqo.decycle.gradle.plugin:1.2.3'
 }
--- buildSrc/src/main/groovy/rhino.library-conventions.gradle.orig	2026-02-16 06:34:26
+++ buildSrc/src/main/groovy/rhino.library-conventions.gradle	2026-09-09 16:09:43
@@ -13,7 +13,7 @@
 }
 
 dependencies {
-    errorprone "com.google.errorprone:error_prone_core:2.28.0"
+    errorprone "com.google.errorprone:error_prone_core:2.45.0"
 }
 
 version = project.version
@@ -34,40 +34,50 @@
 }
 
 tasks.withType(JavaCompile).configureEach {
-    options.errorprone.disable(
-        // JavaDoc stuff
-        "AlmostJavadoc",
-        "EmptyBlockTag",
-        "EscapedEntity",
-        "MissingSummary",
-        "InvalidBlockTag",
-        "NotJavadoc",
-        "UnicodeEscape",
-        // Great ideas for when minimum version is more than Java 11
-        "PatternMatchingInstanceof",
-        // Stuff that we just love to do but should do less of eventually
-        "EmptyCatch",
-        "LabelledBreakTarget",
-        "JavaUtilDate",
-        // Less important for now, more stylistic than bug
-        "InlineMeSuggester",
-        // This one either alerts for parameters that we don't use,
-        // or spuriously for local variables in my opinion.
-        "UnusedVariable",
-        // This requires annotations that are difficult to incorporate
-        // without adding a dependency for everyon who uses Rhino.
-        "AnnotateFormatMethod"
+    if (JavaVersion.current() < JavaVersion.VERSION_17) {
+        // Latest versions of ErrorProne are only compatible with Java 17
+        // and up.
+        System.out.println("Compiling with Java " + JavaVersion.current() +
+          " ErrorProne checks disabled");
+        options.errorprone.enabled = false;
+    } else {
+        // Disable tests that we aren't ready for in ErrorProne.
+        options.errorprone.disable(
+                // Things we should fix as we increase Java versions
+                "ClassInitializationDeadlock",
+                "EffectivelyPrivate",
+                "BooleanLiteral",
+                "AssignmentExpression",
+                "DuplicateBranches",
+                "PatternMatchingInstanceof",
+                "RedundantControlFlow",
+                // JavaDoc stuff -- not going to fix unless we do a big JavaDoc update
+                "AlmostJavadoc",
+                "EmptyBlockTag",
+                "EscapedEntity",
+                "MissingSummary",
+                "InvalidBlockTag",
+                "NotJavadoc",
+                "UnicodeEscape",
+                // Stuff that we just love to do but should do less of eventually
+                "EmptyCatch",
+                "LabelledBreakTarget",
+                "JavaUtilDate",
+                // Less important for now, more stylistic than bug
+                "InlineMeSuggester",
+                // This one either alerts for parameters that we don't use,
+                // or spuriously for local variables in my opinion.
+                "UnusedVariable",
+                // This requires annotations that are difficult to incorporate
+                // without adding a dependency for everyon who uses Rhino.
+                "AnnotateFormatMethod"
         )
-    options.errorprone.enable(
-        // These two do the last thing that Checkstyle used to do for us
-        "RemoveUnusedImports",
-        "WildcardImport",
-        // Warn us proactively that we are using things that might not work on Android.
-        // We will aggressively used "SuppressWarnings" for these, but these can be
-        // a reminder to check with an Android user since we don't have automated
-        // Android unit tests.
-        "AndroidJdkLibsChecker")
-    options.errorprone.excludedPaths = '.+/src/test/java/.+'
+        options.errorprone.enable(
+                // These two do the last thing that Checkstyle used to do for us
+                "RemoveUnusedImports",
+                "WildcardImport")
+        options.errorprone.excludedPaths = '.+/src/test/java/.+'
+    }
 }
 
 tasks.withType(Javadoc) {
--- buildSrc/src/main/groovy/rhino.spotless-conventions.gradle.orig	2026-02-16 06:34:26
+++ buildSrc/src/main/groovy/rhino.spotless-conventions.gradle	2026-09-09 16:12:24
@@ -12,7 +12,7 @@
     // requires at least Java 17.
     if (JavaVersion.current() >= JavaVersion.VERSION_17) {
         java {
-            googleJavaFormat('1.23.0').aosp()
+            googleJavaFormat('1.33.0').aosp()
         }
     } else {
         System.out.println("Not running Spotless: Java language version is " + JavaVersion.current())
--- gradle.properties.orig	2026-02-16 06:34:26
+++ gradle.properties	2026-09-09 16:14:20
@@ -8,3 +8,4 @@
 org.gradle.parallel=true
 org.gradle.configuration-cache=true
 android.useAndroidX=true
+org.gradle.jvmargs=--enable-native-access=ALL-UNNAMED
--- rhino-kotlin/build.gradle.orig	2026-02-16 06:34:26
+++ rhino-kotlin/build.gradle	2026-09-09 16:19:43
@@ -14,6 +14,15 @@
     jvmToolchain(11)
 }
 
+tasks.compileJava {
+    // Kotlin explicitly runs with an older Java version which does not work
+    // with the current ErrorProne.
+    options.errorprone.enabled = false;
+}
+tasks.compileTestJava {
+    options.errorprone.enabled = false;
+}
+
 publishing {
     publications {
         rhinokotlin(MavenPublication) {
--- tests/src/test/java/org/mozilla/javascript/tests/es6/NativeDateTest.java.orig	2026-02-16 06:34:26
+++ tests/src/test/java/org/mozilla/javascript/tests/es6/NativeDateTest.java	2026-09-09 16:23:03
@@ -725,10 +725,13 @@
         toTimeString("22:23:00 GMT-0000 (UTC)", "UTC");
     }
 
+    /*
+    * The time zone that comes back from this platform differs between Java versions,
+    * and ECMAScript says that this functionality may be platform-dependent.
     @Test
     public void toTimeStringEST() {
-        toTimeString("17:23:00 GMT-0500 (-05:00)", "EST");
-    }
+        toTimeString("17:23:00 GMT-0500 (EST)", "EST");
+    }*/
 
     @Test
     public void toTimeStringBerlin() {
